compiler/noirc_frontend/src/hir/comptime/value.rs

```diff
@@ -174,6 +174,47 @@ impl Value {
         interner.push_expr_type(id, typ);
         Ok(id)
     }
+
+    /// Converts any unsigned `Value` into a `u128`.
+    /// Returns `None` for negative integers.
+    pub(crate) fn to_u128(&self) -> Option<u128> {
+        match self {
+            Self::Field(value) => Some(value.to_u128()),
+            Self::I8(value) => {
+                if value < &0 {
+                    None
+                } else {
+                    Some(*value as u128)
+                }
```

Comment:
```suggestion
                (*value >= 0).then(|| *value as u128)
```
Could save some lines here